era_5g_interface/frame_encoder.py

```diff
@@ -9,35 +9,46 @@
 from av.video.frame import VideoFrame
 
 
-class H264EncoderError(Exception):
-    """FFmpegError Exception."""
+class FrameEncoderError(Exception):
+    """FrameEncoderError Exception."""
 
     pass
 
 
 # TODO: only for testing purpose
+# from pathlib import Path
 # Path("input").mkdir(parents=True, exist_ok=True)
 
-logger = logging.getLogger("H.264 encoder")
+logger = logging.getLogger("Video frame encoder")
 
 
-class H264Encoder:
-    """H.264 Encoder."""
+class FrameEncoder:
+    """Video frame Encoder."""
 
-    def __init__(self, width: int, height: int, fps: float = 30, options: Optional[Dict[str, str]] = None) -> None:
+    def __init__(
+        self, width: int, height: int, fps: float = 30, codec: str = "h264", options: Optional[Dict[str, str]] = None
+    ) -> None:
         """Constructor.
 
         Args:
             width (int): Video frame width.
             height (int): Video frame height.
             fps (float): Video framerate (FPS), default: 30.
-            options (Dict[str, str], optional): H264 options, e.g. {"crf": "0", "preset": "ultrafast",
+            codec (str): Video codec name, e.g. h264, hevc, or vp9, default: h264.
+            options (Dict[str, str], optional): Codec options, e.g. {"crf": "0", "preset": "ultrafast",
                 "tune": "zerolatency", "x264-params": "keyint=5"}, default: {"preset": "ultrafast",
                 "tune": "zerolatency"}.
         """
 
+        # Some default options.
         if options is None:
-            options = {"preset": "ultrafast", "tune": "zerolatency"}
+            if codec == "h264":
+                options = {"preset": "ultrafast", "tune": "zerolatency"}
+            elif codec == "hevc":
+                options = {"crf": "10", "preset": "ultrafast", "tune": "zerolatency", "x265-params": "frame-threads=1"}
+            elif codec == "vp9":
```

Comment:
vp9 is specified here but only h264 and HEVC are in ChannelType enum

Reply:
FrameEncoder and FrameDecoder classes are theoretically for any video codec supported by FFmpeg, only these three have been tested and there is a default setting suitable for transmission without delay. I didn't add vp9 to channels because it seemed computationally more demanding, but you can always add it there.

Reply:
ok, no problem